src/csrs/service/civilServantProfileService.ts

```diff
@@ -0,0 +1,29 @@
+import * as config from '../../config/index'
+import { AxiosInstance } from "axios"
+
+export class CivilServantProfileService {
+    http: AxiosInstance
+
+    constructor(http: AxiosInstance){
+        this.http = http
+    }
+
+    async getProfile(accessToken: string){
```

Comment:
This could be incorporated into the regular `csrsService` using the `restService.getWithConfig` function - saves using an extra class

Reply:
The main reason I created a new class is because `OauthRestService` depends on `Auth` (as it's a subclass of `JsonRestService`), making it impossible to call it in the `Auth` service as we'll be going in circles. I created a new `CivilServantProfileService` class which is just needs `AxiosInstance`, which completely decouples `Auth`.